### PR TITLE
Doorkeeper access tokens can be JWT tokens

### PR DESCRIPTION
**Pull request by marten**
This allows auxiliary services (Nero, EducationAPI, ZooEventStats, ...) to independently verify the validity of OAuth tokens sent to them by clients. Right now they have to make an API call to Panoptes to verify any tokens given to them (`GET /api/me`).

Currently issued tokens will expire at some point (a few hours) and get replaced with JWT-based ones.

### CODE REVIEW

**Comment by marten** [comment]
@zooniverse/developers Anyone foresee any issues with the OAuth token string (the bit in the `Authorization: Bearer TOKENHERE` header) becoming a bit longer (around 500 chars instead of 64 chars)? I'm thinking it should be fine as long as it fits inside a cookie?

**Comment by parrish** [comment]
I don't see any issues with a longer auth string.  👍 for introducing JWT too.

**Comment by camallen** [comment]
Doesn't seem like there will be problems...seems there are size limits on headers for web servers but it's pretty big

**Comment by marten** [comment]
Can't think of any server-side applications that might be storing access tokens either (which might have too small a DB-column).

File: spec/support/controller_helpers.rb [hunk 1]
```diff
@@ -42,7 +42,11 @@ def set_preconditions
   end
 
   def stub_token(scopes: [], user_id: nil)
-    allow(controller).to receive(:doorkeeper_token).and_return(token(scopes, user_id))
+    if user_id
+      allow(controller).to receive(:doorkeeper_token).and_return(token(scopes, user_id))
+    else
+			allow(controller).to receive(:doorkeeper_token).and_return(nil)
```
Comment (camallen) [inline spec/support/controller_helpers.rb]: what's up with the spacing?

**Comment by camallen** [comment]
LGTM

**Pull request merged**